homeassistant/components/dremel_3d_printer/entity.py

```diff
@@ -0,0 +1,41 @@
+"""Entity representing a Dremel 3D Printer."""
+
+from dremel3dpy import Dremel3DPrinter
+
+from homeassistant.helpers.entity import DeviceInfo, EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import Dremel3DPrinterDataUpdateCoordinator
+
+
+class Dremel3DPrinterEntity(CoordinatorEntity[Dremel3DPrinterDataUpdateCoordinator]):
+    """Defines a Dremel 3D Printer device entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: Dremel3DPrinterDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the base device entity."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_{description.key}"
```

Comment:
You should use serial, not entry ID.
```suggestion
        self._attr_unique_id = f"{coordinator.config_entry.unique_id}_{description.key}"
```